src/uu/expr/src/syntax_tree.rs

```diff
@@ -286,8 +302,28 @@ where
     }
 
     // Check if parsed quantifier is valid
-    let re = Regex::new(r"(\d+|\d*,\d*)").expect("valid regular expression");
-    re.is_match(&quantifier)
+    let re = Regex::new(r"^(\d*,\d*|\d+)").expect("valid regular expression");
+    match re.captures(&quantifier) {
```

Comment:
I would use an `if let` instead of a `match` as it allows you to remove one indentation level.